pkgs/servers/onlyoffice-documentserver/default.nix

```
@@ -17,7 +17,10 @@ let
     pname = "onlyoffice-documentserver";
     version = "7.4.0";
 
-    src = fetchurl {
+    src = if stdenv.hostPlatform.system == "aarch64-linux" then fetchurl {
+      url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version))}/onlyoffice-documentserver_arm64.deb";
+      sha256 = "sha256-JHDdwLAfrH+hKF0c1UqcjwxTdeYFjWVEjzjG8HTcqaA=";
+    } else fetchurl {
       url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version))}/onlyoffice-documentserver_amd64.deb";
       sha256 = "sha256-FL09EXxQlUZuJMMHYu9tSOH8ARPgzoqAKmQYV6225PU=";
     };
```

```suggestion
    src = let
      system = stdenv.hostPlatform.system;
      tag = lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version;
    in {
      "x86_64-linux" = fetchurl {
        url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${tag}/onlyoffice-documentserver_arm64.deb";
        hash = "sha256-FL09EXxQlUZuJMMHYu9tSOH8ARPgzoqAKmQYV6225PU=";
      };
      "aarch64-linux" = fetchurl {
        url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${tag}/onlyoffice-documentserver_amd64.deb";
        hash = "sha256-JHDdwLAfrH+hKF0c1UqcjwxTdeYFjWVEjzjG8HTcqaA=";
      };
    }.${system} or throw "unsupported system: ${system}"
```
we shouldn't if else daisy chain this

---

@SuperSandro2000 I agree completely, but it is only those two system architectures being supported at the moment and the only other example I could find is the following [package / commit](https://github.com/NixOS/nixpkgs/commit/39f4c0ae138d432a8b5ca77cf804e38ad456e9af).